server/src/main/java/io/spine/server/commandbus/CommandAcknowledgementMonitor.java

```diff
@@ -0,0 +1,153 @@
+
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+
+/**
+ * A {@link StreamObserver} for {@link io.spine.core.Command Command}
+ * {@linkplain Ack acknowledgement}.
+ *
+ * <p>Posts a system command whenever a command is acknowledged or errored.
+ *
+ * @author Dmytro Dashenkov
+ */
+final class CommandAcknowledgementMonitor implements StreamObserver<Ack> {
+
+    private final StreamObserver<Ack> delegate;
+    private final TenantId tenantId;
+
+    private final SystemGateway gateway;
+
+    private CommandAcknowledgementMonitor(Builder builder) {
+        this.delegate = builder.delegate;
+        this.tenantId = builder.tenantId;
+        this.gateway = builder.systemGateway;
+    }
+
+    @Override
+    public void onNext(Ack value) {
+        delegate.onNext(value);
+        postSystemCommand(value);
+    }
+
+    @Override
+    public void onError(Throwable t) {
+        delegate.onError(t);
+    }
+
+    @Override
+    public void onCompleted() {
+        delegate.onCompleted();
+    }
+
+    private void postSystemCommand(Ack ack) {
+        Status status = ack.getStatus();
+        CommandId commandId = commandIdFrom(ack);
+        Message systemCommand = systemCommandFor(status, commandId);
+        gateway.postCommand(systemCommand, tenantId);
+    }
+
+    private static CommandId commandIdFrom(Ack ack) {
+        Any messageId = ack.getMessageId();
+        return Identifier.unpack(messageId);
+    }
+
+    @SuppressWarnings("EnumSwitchStatementWhichMissesCases") // Default values.
+    private static Message systemCommandFor(Status status, CommandId commandId) {
+        switch (status.getStatusCase()) {
+            case OK:
+                return AcknowledgeCommand.newBuilder()
+                                         .setId(commandId)
+                                         .build();
+            case ERROR:
+                return MarkCommandAsErrored.newBuilder()
+                                           .setId(commandId)
+                                           .setError(status.getError())
+                                           .build();
+            case REJECTION:
+            default:
+                throw newIllegalArgumentException("Invalid status %s.", status.getStatusCase());
+        }
+    }
+
+    /**
+     * Creates a new instance of {@code Builder} for {@code CommandAcknowledgementMonitor} instances.
+     *
+     * @return new instance of {@code Builder}
+     */
+    static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A builder for the {@code CommandAcknowledgementMonitor} instances.
+     */
+    static final class Builder {
+
+        private StreamObserver<Ack> delegate;
+        private TenantId tenantId;
+        private SystemGateway systemGateway;
+
+        /**
+         * Prevents direct instantiation.
+         */
+        private Builder() {
+        }
+
+        Builder setDelegate(StreamObserver<Ack> delegate) {
```

Comment:
Let's do three things:

1. Document the builder params. Honestly, I have no idea what this delegate is.
2. Check the args for not being `null`s.
3. Test.